hladnymatfyzak/__init__.py

```diff
@@ -90,20 +90,33 @@ def dolna(day=None, month=None, year=None):
         return list
 
     req = requests.get(lin)
-    soup = BeautifulSoup(req.text)
-    tables = soup.find_all('table')
-    meals_table = tables[weekday*9]
-    soup_table = tables[weekday*9 + 6]
-    meals = re.findall(r'<td width="400">(.*?)</td>', str(meals_table) )
-    meal_price = re.findall(r'<td>(.*?)............./', str(meals_table))
-    soups = re.findall(r'<td width="400">(.*?)</td>',str(soup_table))
-    soup_price = re.findall(r'<td>(.*?)............./', str(soup_table))
+    parts = req.text.split("<a name='")
+    datestr = "{0}-{1:02d}-{2:02d}".format(year,month,day)
+    daymenu = None
+    for part in parts:
+        if part.startswith(datestr):
+            daymenu = part
+            
+    if daymenu == None:
+        return ["menu for given day not found"]
+    
+    daymenu_bs = BeautifulSoup(str(daymenu))
+    tables = daymenu_bs.find_all('table')
+    meals = re.findall(r'<td width="400">(.*?)</td>', str(tables[0]))
+    meal_price = re.findall(r'<td>..........................(.*?).....\*', str(tables[0]))
+    soups = re.findall(r'<td width="400">(.*?)</td>',str(tables[6]))
```

Comment:
@tWido in all these findalls you nicely use a space after the comma. Why not here?